axelrod/graph.py

```diff
@@ -0,0 +1,126 @@
+"""
+Weighted undirected sparse graphs.
+
+Original source:
+https://github.com/marcharper/stationary/blob/master/stationary/utils/graph.py
```

Comment:
> There are cases however that a list of edges doesn't cover, like some types of random graphs.

That's the killer point right there! I guess that also includes the possibility of graphs that change over time etc which also rules out networkX (which really is a data representation library coupled with graph theoretic algorithms as opposed to the flexibility we might need here).

I suggest going for flexibility on the input (so we can open an issue for my suggestion of `3.i`).